src/west/app/project.py

```diff
@@ -640,6 +652,68 @@ def _die_if_manifest_project_filter(self, action):
                      'project filter.')
 
+    def _untracked(self):
+        ''' "Performs a top-down search of the west topdir,
+        ignoring every folder that corresponds to a west project.
+        '''
+        ppaths = []
+        untracked = []
+        for project in self._projects(None):
+            # We do not check for self.manifest.is_active(project) because
+            # inactive projects are still considered "tracked folders".
+            ppaths.append(Path(project.abspath))
+
+        # Since west tolerates nested projects (i.e. a project inside the folder
+        # of another project) we must sort the project paths to ensure that we
+        # hit the "enclosing" project first when iterating.
+        ppaths.sort()
+
+        def _find_untracked(folder):
+            '''There are three cases for each element in a folder:
+            - It's a project -> Do nothing, ignore the folder.
+            - There are no projects inside -> add to untracked list.
+            - It's not a project folder but there are some projects inside it -> recurse.
+            The folder argument cannot be inside a project, otherwise all bets are off.
+            '''
+            self.dbg(f'looking for untracked files/folders in: {folder}')
+            for e in [e.absolute() for e in folder.iterdir()]:
+                if not e.is_dir() or e.is_symlink():
+                    untracked.append(e)
+                    continue
+                self.dbg(f'processing folder: {e}')
+                for ppath in ppaths:
+                    # We cannot use samefile() because it requires the file
+                    # to exist (not always the case with inactive or even
+                    # uncloned projects).
+                    if ppath == e:
+                        # We hit a project root folder, skip it.
+                        break
+                    elif e in ppath.parents:
+                        self.dbg(f'recursing into: {e}')
+                        _find_untracked(e)
+                        break
+                else:
+                    # This is not a project and there is no project inside.
+                    # Add to untracked elements.
+                    untracked.append(e)
+                    continue
+
+        # Avoid using Path.walk() since that returns all files and folders under
+        # a particular folder, which is overkill in our case. Instead, recurse
+        # only when required.
+        _find_untracked(Path(self.topdir))
+
+        # Remove the .west folder, which is maintained by west
+        try:
+            untracked.remove((Path(self.topdir) / Path(WEST_DIR)).resolve())
+        except ValueError:
+            self.die(f'Folder {WEST_DIR} not found in workspace')
+
+        # Sort the results for displaying to the user.
+        untracked.sort()
+        for u in untracked:
+            self.inf(u.relative_to(Path.cwd(), walk_up=True))
```

Comment:
`walk_up` was added in Python 3.12 and will not work on older versions.

Reply:
Good catch! I now have to find an alternative.

Reply:
Fixed, thanks @pdgendt